src/gnn_reco/components/loss_functions.py

```diff
@@ -73,12 +73,12 @@ def _log_cosh(cls, x: Tensor) -> Tensor:  # pylint: disable=invalid-name
         Used to avoid `inf` for even moderately large differences.
         See [https://github.com/keras-team/keras/blob/v2.6.0/keras/losses.py#L1580-L1617]
         """
-        return x + torch.nn.functional.softplus(-2. * x) - np.log(2.0)
+        return torch.log(torch.cosh(x))#x + torch.nn.functional.softplus(-2. * x) - np.log(2.0)
 
     def _forward(self, prediction: Tensor, target: Tensor) -> Tensor:
         """Implementation of loss calculation."""
         assert prediction.dim() == target.dim() + 1
-        diff = prediction[:,0] - target
+        diff = prediction[:,0] - torch.log10(target)
```

Comment:
I really don't think we should hard-code a log10 transform into a generic LogCosh loss. This is specific to energy regression, and so I would propose this transform be imposed only there, e.g. as:
```
task = EnergyReconstruction(
    loss_function=LogCoshLoss(
        transform_output=lambda x: torch.log10(x),
    ),
    ...
)
```

Reply:
This is also one of the changes that leads to a failing unit test.